### PR TITLE
[MRG] Progress bars

### LINKED ISSUE #55: File sizes are zero
`osf clone` produces the right file tree but all files have zero size.

python 3, current master.

Looking into it at the moment.

### PR DESCRIPTION
**Pull request by betatim**
Fix #55 and adds progres bars.

This is a bit of a mishmash as it adds new features and fixes a bug.

Need to think about what test would have caught the streaming bug.

* [x] test for streaming bug
* [ ] test progress bars (??)
* [x] test that `osf fetch` stops after it found the file it was looking for

### CODE REVIEW

File: osfclient/models/file.py [hunk 1]
```diff
@@ -35,10 +49,11 @@ def write_to(self, fp):
         if 'b' not in fp.mode:
             raise ValueError("File has to be opened in binary mode.")
 
-        response = self._get(self._download_url)
+        response = self._get(self._download_url, stream=True)
```
Comment (betatim) [inline osfclient/models/file.py]: This is the fix for #55

**Comment by betatim** [comment]
Current status: this whole `mock` thing seemed like a good idea at the time but right now thinking it is far too complex :-/

**Comment by betatim** [comment]
This adds progress bars for file download :tada: (`osf fetch` and `osf clone`) not sure how to add a unit test for the progress bar so punting on that now.

Let's make a new PR to add progress bars for uploading.

This also fixes a few bugs found along the way. If there are suggestions (-> new issue or #39) for how we can simplify the test setup without having to make HTTP requests that would be good.

Ready for review.

**Comment by betatim** [comment]
One thing that is missing from this PR is to make the progress bars optional. Maybe something for a second PR.

**Comment by gedankenstuecke** [comment]
Ok, gotta finish this PR for openSNP and then I'll go through it. 🍰

**Comment by betatim** [comment]
:tada: paging Dr @ctb to the progress bar

**Pull request merged**